src/main/java/touro/alter/GraphCreator.java

```diff
@@ -0,0 +1,54 @@
+package touro.alter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.*;
+
+/**
+ * reads input from text file and creates connections between nodes to make "graph"
+ */
+public class GraphCreator {
+
+   private List<GraphNode> nodeList = new ArrayList<>();
+
+    public GraphCreator(String file){
+        try {
+            Scanner inputFile = new Scanner(new File(file));
+            while(inputFile.hasNextLine()) {
+                String input = inputFile.nextLine();
+
+                connectNodes(input);
+
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    private void connectNodes(String input) {
+        if (!input.contentEquals("0")) {
+           String vertices[] = input.split(" ");
+
+           GraphNode firstNode = new GraphNode(vertices[0]);
```

Comment:
I think creating new graphnode objects here creates a bug where if the node DOES already exist in the list (ie the node's value exists), it does not add a connection to the existing node because it has a new address